doc/source/whatsnew/v1.2.0.rst

```diff
@@ -96,6 +96,20 @@ For example:
    buffer = io.BytesIO()
    data.to_csv(buffer, mode="w+b", encoding="utf-8", compression="gzip")
 
+Inverting a misaligned mask within ``where``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+:meth:`Series.where` and :meth:`DataFrame.where` now include an argument ``invert`` to handle inversion of a possibly misaligned mask. The inversion is done after realignment so as to properly handle any missing values.
+
+.. ipython:: python
+
+    s = pd.Series(range(5), dtype=float)
+    mask = (s > 2)[:-1]
+    mask
+    s.where(mask)
+    s.where(~mask)
+    s.where(mask, invert=True)
```

Comment:
oh woa, I wouldn't add to the public api at all, maybe my comment wasn't clear this should be for `_where` *only*

Reply:
Moved this argument out of the public `where`.

I'm having to do a bit of boolean voodoo to get this to work. Had to bypass a call to `where` and go straight to `_where` which meant moving a small piece of logic that applies a callable "other" there as well. This broke some tests because apparently we sometimes we want to do this when using mask / where / setitem and other times not, so needed more control over whether this will happen.